scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala

```diff
@@ -46,13 +49,43 @@ object TinyJoinAndMergeJob {
 class TinyJoinAndMergeJob(args: Args) extends Job(args) {
   import TinyJoinAndMergeJob._
 
-  val people = peopleInput.read.mapTo(0 -> 'id) { v: Int => v }
+  val input1 = joinInput1.read.mapTo(0 -> 'id) { v: Int => v }
+
+  val joinedData = joinInput2.read
+    .mapTo(0 -> 'id) { v: Int => v }
+    .joinWithTiny('id -> 'id, input1)
+
+  val mergerData = mergerInput.read.mapTo(0 -> 'id) { v: Int => v }
+
+  (mergerData ++ joinedData).groupBy('id) { _.size('count) }.write(output)
+}
+
+class TinyJoinAndMergeUnsupportedJob(args: Args) extends Job(args) {
+  import TinyJoinAndMergeJob._
+
+  val input1 = joinInput1.read.mapTo(0 -> 'id) { v: Int => v }
+
+  val joined = joinInput2.read
+    .mapTo(0 -> 'id) { v: Int => v }
+    .joinWithTiny('id -> 'id, input1)
+
+  // merging the output of a hashjoin with one of its inputs is
+  // no longer supported in cascading3. So we verify we fail here.
+  (joined ++ input1).groupBy('id) { _.size('count) }.write(output)
+}
+
+class TinyJoinAndMergeForceToDiskJob(args: Args) extends Job(args) {
+  import TinyJoinAndMergeJob._
+
+  val input1 = joinInput1.read.mapTo(0 -> 'id) { v: Int => v }
 
-  val messages = messageInput.read
+  val joined = joinInput2.read
     .mapTo(0 -> 'id) { v: Int => v }
-    .joinWithTiny('id -> 'id, people)
+    .joinWithTiny('id -> 'id, input1)
+    .forceToDisk // workaround for cascading3
```

Comment:
Looks like there is no way to access the underlying Pipes in a Merge right now. The closest is getPipePos which returns (name, position) pairs:
https://github.com/cwensel/cascading/blob/wip-3.1/cascading-core/src/main/java/cascading/pipe/Splice.java#L944

I've sent an email to cascading-user list to look into adding a getPipes method. We may have to rely on the name otherwise.